### ISSUE TITLE
Xcode 15.3 / Swift 5.10 is not supported yet.

### ISSUE THREAD

**Issue by soranoba** [Quick/Quick#1275]
- [x] I have read [CONTRIBUTING](https://github.com/Quick/Quick/blob/main/CONTRIBUTING.md) and have done my best to follow them.

### What did you do?

I upgraded to xcode15.3.

### What did you expect to happen?

Test can be executed.

### What actually happened instead?

It failed to compile with Cachege `--no-use-binaries` option.

```
missing required module 'CwlCatchExceptionSupport'
```

### Environment

List the software versions you're using:

 - Quick: *7.4.1*
 - Nimble: *13.2.1*
 - Xcode Version: *15.3* (Open Xcode; In menubar: Xcode > About Xcode)
 - Swift Version: *5.10* (Open Xcode Preferences; Components > Toolchains. If none, use `Xcode Default`.)

Please also mention which package manager you used and its version. Delete the
other package managers in this list:

 - Carthage: *0.38.0* (Use `carthage version` in Terminal)

### Project that demonstrates the issue

Please link to a project we can download that reproduces the issue. Feel free
to delete this section if it's not relevant to the issue (eg - feature request).

The project should be [short, self-contained, and correct example](http://sscce.org/).

**Comment by soranoba** [comment]
If I use following versions and `--no-use-binaries`, it compile successded. 🤔 

github "Quick/Nimble" "v13.0.0"
github "Quick/Quick" "v7.3.0"

**Comment by younata** [comment]
In Nimble 13.1 (December 2023), we switched to pull in CwlPreconditionTesting transitively using SPM (https://github.com/Quick/Nimble/pull/1108). This works in Carthage 0.39.1 (released in September 2023), and Xcode 15.3:

```
$ carthage update --no-use-binaries --use-xcframeworks
*** Fetching Nimble
*** Fetching Quick
*** Checking out Quick at "v7.4.1"
*** Checking out Nimble at "v13.2.1"
*** xcodebuild output can be found in /var/folders/_v/6_5lvql10ks78qc_3r52m7500000gn/T/carthage-xcodebuild.60frDa.log
*** Building scheme "Nimble" in Nimble.xcodeproj
*** Building scheme "Quick" in Quick.xcworkspace
$ echo $?
0
```

I haven't checked if this is broken when using older versions of Carthage, but maybe that's what's going on?

**Comment by soranoba** [comment]
The compile error occurs when I run the test in Xcode. (carthage build successeded)
I will try increasing the version of carthage and check it.

**Comment by younata** [comment]
Oh. Actually, after trying to embed it in a test app, it looks like there's some bug in... somewhere? Xcode? Carthage? Nimble's configuration? causing CWLPreconditionTesting to not be embedded? You can work around this by adding the package manually to your project (the URL is https://github.com/mattgallagher/CwlPreconditionTesting, and you'll need version 2.2.0 or later). What a terrible experience. I'm sorry.

**Comment by soranoba** [comment]
I think the reasons is that dependencies are not written in Cartfile.
https://github.com/Quick/Nimble/blob/256e242d294b691103305588cbab176ed617a39e/Cartfile.resolved

**Comment by younata** [comment]
Yes, because those dependencies no longer support Carthage. Nimble 13 and earlier were using an older version of those testing aids from when they did support Carthage.

**Comment by soranoba** [comment]
Oh....

**Comment by soranoba** [comment]
The title has been changed.

**Comment by CraigSiemens** [comment]
We ran into this as well. Currently we're just using Nimble 13.0 since we're in no rush to update.

It's possibly it's caused by Xcode automatically deciding to dynamically link the `Cwl*` packages when building nimble. AKAIK the only solution would be for those dependencies to define static products that can be explicitly chosen by Nimble.

**Comment by CraigSiemens** [comment]
I tried the approach of manually adding the `CwlPreconditionTesting` package to the project. 

It fixed the build but added a warning at runtime
```
objc[32817]: Class BadInstructionException is implemented in both 
.../Build/Products/Debug-iphonesimulator/Nimble.framework/Nimble (0x107cb3860) and 
.../Build/Products/Debug-iphonesimulator/TestTarget.framework/TestTarget (0x103494208).
This may cause spurious casting failures and mysterious crashes. One of the duplicates must be removed or renamed.
```

It also caused the tests that use the `throwsAssertion()` to randomly fail.
